### ISSUE TITLE
[BUG] Group chat automatically trimming when not supposed to.

### ISSUE THREAD

**Issue by imesha10** [SillyTavern/SillyTavern#1014]
**Have you searched for similar [bugs](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes.

**Describe the bug**

https://github.com/SillyTavern/SillyTavern/assets/16441743/cb01d049-091d-458b-87fb-8cb7442593f6

I have a prompt that asks at the bottom asks the user to print info regarding the scene. One of the things it prints is the location and it prints it out like this: `User: home, Char1: living room, Char2: basement`. 
What happens while the text is being streamed is first `User: home, Char1: living room, Char2:` is generated and it stops and becomes `User: home, Char1: living room`.
The timer still keeps running after.

Also I don't have any regexes or anything else that cuts this automatically turned on. I even turned on all the show char: options in the settings that I can find.

**To Reproduce**
1. Go to a group chat with at least 2 members.
2. Write a prompt that get's it to print another character in the for of `char:`
3. See when it stops.

**Expected behavior**
Have the option somewhere to stop the automatic cuttoff that happens when it detects 'char:'.

**Logs**

<details>
<summary>Expand to see Chrome logs.</summary>

``` shell
moment.js:287 Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
Arguments: 
[0] _isAMomentObject: true, _isUTC: false, _useUTC: false, _l: undefined, _i: August 24, 2023 10:40pm, _f: undefined, _strict: undefined, _locale: [object Object]
Error
    at Function.createFromInputFallback (http://localhost:8000/lib/moment.min.js:1:3257)
    at Nt (http://localhost:8000/lib/moment.min.js:1:21587)
    at Pt (http://localhost:8000/lib/moment.min.js:1:22164)
    at W (http://localhost:8000/lib/moment.min.js:1:22247)
    at f (http://localhost:8000/lib/moment.min.js:1:207)
    at timestampToMoment (http://localhost:8000/scripts/utils.js:493:9)
    at addOneMessage (http://localhost:8000/script.js:1407:24)
    at sendSystemMessage (http://localhost:8000/script.js:1839:5)
    at Object.<anonymous> (http://localhost:8000/script.js:693:5)
    at e (http://localhost:8000/lib/jquery-3.5.1.min.js:2:30005)
B @ moment.js:287
(anonymous) @ moment.js:319
Nt @ moment.js:2550
Pt @ moment.js:2943
W @ moment.js:3041
f @ moment.js:16
timestampToMoment @ utils.js:493
addOneMessage @ script.js:1407
sendSystemMessage @ script.js:1839
(anonymous) @ script.js:693
e @ jquery-3.5.1.min.js:2
t @ jquery-3.5.1.min.js:2
setTimeout (async)
(anonymous) @ jquery-3.5.1.min.js:2
c @ jquery-3.5.1.min.js:2
fireWith @ jquery-3.5.1.min.js:2
fire @ jquery-3.5.1.min.js:2
c @ jquery-3.5.1.min.js:2
fireWith @ jquery-3.5.1.min.js:2
l @ jquery-3.5.1.min.js:2
(anonymous) @ jquery-3.5.1.min.js:2
load (async)
send @ jquery-3.5.1.min.js:2
ajax @ jquery-3.5.1.min.js:2
S.<computed> @ jquery-3.5.1.min.js:2
(anonymous) @ script.js:690
openai.js:2560 OpenAI model changed to gpt-4-0314
power-user.js:421 applying bubblechat
elevenlabs.js:53 Using default TTS Provider settings
elevenlabs.js:73 Settings loaded
thumbnail:1     GET http://localhost:8000/thumbnail?type=avatar&file=Terra.png 404 (Not Found)
index.js:202 No current task
script.js:2292 No enabled members found
script.js:2410 Core/all messages: 2/4
script.js:2949 pushed prompt bits to itemizedPrompts array. Length is now: 1
2script.js:2292 No enabled members found
script.js:2410 Core/all messages: 2/4
script.js:2949 pushed prompt bits to itemizedPrompts array. Length is now: 2
2script.js:2292 No enabled members found
```
</details>

**Settings screenshot.**
![image](https://github.com/SillyTavern/SillyTavern/assets/16441743/bf2e8b0b-7ce0-43f4-baa8-8d03bac42e73)
![image](https://github.com/SillyTavern/SillyTavern/assets/16441743/c81bf1db-982b-40c2-aef8-ad46682e0845)
![image](https://github.com/SillyTavern/SillyTavern/assets/16441743/1c19d2e1-db64-48a9-9954-8f1ca9eef0fb)

**Desktop (please complete the following information):**
 - OS/Device: windows 11
 - Environment: local
 - Node.js version (if applicable): v18.15.0
 - Browser: chrome
 - Generation API: openAI
 - Branch: staging
 - Model: gpt3.5/gpt4

**Comment by Cohee1207** [comment]
Sorry, this was added to prevent group members from speaking as other characters in the same chat, but I didn't accommodate mentioning name + colon in the text. I fixed the name trimming in the following way: only trim the text if it's the very start of the message or it has a preceding new line. Basically, the same way name trimming works for solo chats.

After the fix it will allow things like this:
![image](https://github.com/SillyTavern/SillyTavern/assets/18619528/26a12aff-8fa9-4482-9716-6b21c6c5e871)
